### PR TITLE
Add golanci-lint and build CI to project

### PR DESCRIPTION
**Pull request by abdallahyas**
Added the following CIs:
 - golangci-lint: A github workflow to run golanci-lint against the project. It uses [1], the Official golangci-lint GitHub action
 - sample-build: A CI to assure package importing is not broken, and that it is possible to build the package used by go scripts. The CI tries to build `ci/sample.go`, if it is successful, then the CI can be considered as success

[1] https://github.com/golangci/golangci-lint-action
Signed-off-by: Abdallah Yasin <abdallahyas@nvidia.com>

### CODE REVIEW

**Comment by abdallahyas** [comment]
@moshe010 @almaslennikov Can you guys PTAL?
Thanks

File: .github/workflows/pr-ci.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+          version: v1.49.0
+
+  build-code:
+    name: sample-build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Prepare GO env
+        uses: actions/setup-go@v4
+        with:
+          go-version: '1.17'
```
Comment (almaslennikov) [inline .github/workflows/pr-ci.yaml]: nit: Shouldn't we use a newer Go version?

**Review by almaslennikov — APPROVED** [review]

File: .github/workflows/pr-ci.yaml (hunk 1 above)
Comment (moshe010) [reply .github/workflows/pr-ci.yaml]: yes, let update it to 1.19 same in go mod. 
https://github.com/Mellanox/mlxdevm-go/blob/main/go.mod#L3

Comment (abdallahyas) [reply .github/workflows/pr-ci.yaml]: Done

**Review by moshe010 — APPROVED** [review]

**Pull request merged**